x/cash/msg.go

```diff
@@ -30,26 +30,22 @@ func (SendMsg) Path() string {
 
 // Validate makes sure that this is sensible
 func (s *SendMsg) Validate() error {
+	var err error
 	if coin.IsEmpty(s.Amount) || !s.Amount.IsPositive() {
-		return errors.Wrapf(errors.ErrAmount, "non-positive SendMsg: %#v", s.Amount)
-
-	}
-	if err := s.Amount.Validate(); err != nil {
-		return errors.Wrap(err, "amount")
-	}
-	if err := s.Src.Validate(); err != nil {
-		return errors.Wrap(err, "src")
-	}
-	if err := s.Dest.Validate(); err != nil {
-		return errors.Wrap(err, "dst")
+		err = errors.Wrapf(errors.ErrAmount, "non-positive SendMsg: %#v", s.Amount)
+	} else {
+		err = errors.Append(err, s.Amount.Validate())
 	}
+	err = errors.Append(err, s.Src.Validate())
+	err = errors.Append(err, s.Dest.Validate())
```

Comment:
In other news, I think the use of Append here is actually cleaner than the if/return logic.
Just make sure Append and Wrap pass `nil` input errors through without changes, and it just flows